prow/config/inrepoconfig.go

```diff
@@ -0,0 +1,126 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/git"
+	"k8s.io/test-infra/prow/github"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	inRepoConfigFileName = ".prow.yaml"
+)
+
+// ProwYAML represents the content of a .prow.yaml file
```

Comment:
nit: not sure if we care but technically with the way this is written today I could have a JSON file and it would load fine, not sure we need to call out the storage mechanism in the name of things

Reply:
It was primarily named like that because the file is named like that and that can not be different. But totally open for a better name if you have a suggestion

Reply:
Sure -- if I put JSON into my `prow.yaml` it will work ;)